### PR TITLE
[net7.0] [Windows] Fix CollectionView.RemainingItemsThresholdReached

### PR DESCRIPTION
**Pull request by hartez**
Backport of #14391 to net7.0

/cc @hartez @Foda

### CODE REVIEW

**Review by rmarinho — CHANGES_REQUESTED** [review]
```
D:\a\_work\1\s\src\Controls\tests\DeviceTests\Elements\CollectionView\CollectionViewTests.Windows.cs(94,13): error CS0246: The type or namespace name 'WSetter' could not be found (are you missing a using directive or an assembly reference?) [D:\a\_work\1\s\src\Controls\tests\DeviceTests\Controls.DeviceTests.csproj::TargetFramework=net7.0-windows10.0.19041]
D:\a\_work\1\s\src\Controls\tests\DeviceTests\Elements\CollectionView\CollectionViewTests.Windows.cs(94,6): error CS1061: 'SetterBaseCollection' does not contain a definition for 'OfType' and no accessible extension method 'OfType' accepting a first argument of type 'SetterBaseCollection' could be found (are you missing a using directive or an assembly reference?) [D:\a\_work\1\s\src\Controls\tests\DeviceTests\Controls.DeviceTests.csproj::TargetFramework=net7.0-windows10.0.19041]
D:\a\_work\1\s\src\Controls\tests\DeviceTests\Elements\CollectionView\CollectionViewTests.Windows.cs(97,4): error CS0103: The name 'Assert' does not exist in the current context [D:\a\_work\1\s\src\Controls\tests\DeviceTests\Controls.DeviceTests.csproj::TargetFramework=net7.0-windows10.0.19041]
D:\a\_work\1\s\src\Controls\tests\DeviceTests\Elements\CollectionView\CollectionViewTests.Windows.cs(133,5): error CS0103: The name 'Assert' does not exist in the current context [D:\a\_work\1\s\src\Controls\tests\DeviceTests\Controls.DeviceTests.csproj::TargetFramework=net7.0-windows10.0.19041]
D:\a\_work\1\s\src\Controls\tests\DeviceTests\Elements\CollectionView\CollectionViewTests.Windows.cs(94,13): error CS0246: The type or namespace name 'WSetter' could not be found (are you missing a using directive or an assembly reference?) [D:\a\_work\1\s\src\Controls\tests\DeviceTests\Controls.DeviceTests.csproj::TargetFramework=net7.0-windows10.0.20348]
D:\a\_work\1\s\src\Controls\tests\DeviceTests\Elements\CollectionView\CollectionViewTests.Windows.cs(94,6): error CS1061: 'SetterBaseCollection' does not contain a definition for 'OfType' and no accessible extension method 'OfType' accepting a first argument of type 'SetterBaseCollection' could be found (are you missing a using directive or an assembly reference?) [D:\a\_work\1\s\src\Controls\tests\DeviceTests\Controls.DeviceTests.csproj::TargetFramework=net7.0-windows10.0.20348]
D:\a\_work\1\s\src\Controls\tests\DeviceTests\Elements\CollectionView\CollectionViewTests.Windows.cs(97,4): error CS0103: The name 'Assert' does not exist in the current context [D:\a\_work\1\s\src\Controls\tests\DeviceTests\Controls.DeviceTests.csproj::TargetFramework=net7.0-windows10.0.20348]
D:\a\_work\1\s\src\Controls\tests\DeviceTests\Elements\CollectionView\CollectionViewTests.Windows.cs(133,5): error CS0103: The name 'Assert' does not exist in the current context [D:\a\_work\1\s\src\Controls\tests\DeviceTests\Controls.DeviceTests.csproj::TargetFramework=net7.0-windows10.0.20348]
  ```

**Review by jfversluis — APPROVED** [review]

**Review by PureWeen — CHANGES_REQUESTED** [review]

File: src/Controls/tests/DeviceTests/Elements/CollectionView/CollectionViewTests.Windows.cs [hunk 1]
```diff
@@ -49,5 +52,89 @@ public async Task CollectionViewHandlerDisconnects()
 				return Task.CompletedTask;
 			});
 		}
+
+		[Fact]
+		public async Task ValidateItemContainerDefaultHeight()
```
[the remark this answers is not in the document]
Comment (Foda) [reply src/Controls/tests/DeviceTests/Elements/CollectionView/CollectionViewTests.Windows.cs:57]: `ValidateItemContainerDefaultHeight` wasn't part of my PR -- was this added by accident?

**Comment by Foda** [comment]
> Closing this, the original PR on main doesn't work either. No point in backporting this one, we can look at backporting again once it works on main.

@hartez @PureWeen Why on earth do our tests not report failures???
https://github.com/dotnet/maui/pull/14391/checks?check_run_id=13245158008

**Pull request closed without merging**